### ISSUE TITLE
 Broken with storybook@8.2

### ISSUE THREAD

**Issue by yf-yang** [rspack-contrib/storybook-rsbuild#35]
Reproduction link: https://github.com/yf-yang/rsbuild-storybook-esm/tree/sb8.2

``` shell
pnpm storybook
```

```
Error: Cannot find module '@storybook/preview/globals'
Require stack:
- ./node_modules/.pnpm/storybook-builder-rsbuild@0.0.6_@rsbuild+core@1.0.0-alpha.7_@rspack+core@0.7.4_@swc+helpers@0_pealok2q6itvioioojy46f42v4/node_modules/storybook-builder-rsbuild/dist/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._resolveFilename (./node_modules/.pnpm/esbuild-register@3.5.0_esbuild@0.21.5/node_modules/esbuild-register/dist/node.js:4799:36)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (./node_modules/.pnpm/storybook-builder-rsbuild@0.0.6_@rsbuild+core@1.0.0-alpha.7_@rspack+core@0.7.4_@swc+helpers@0_pealok2q6itvioioojy46f42v4/node_modules/storybook-builder-rsbuild/dist/index.js:132:22)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Object.newLoader (./node_modules/.pnpm/esbuild-register@3.5.0_esbuild@0.21.5/node_modules/esbuild-register/dist/node.js:2262:9)
    at extensions..js (./node_modules/.pnpm/esbuild-register@3.5.0_esbuild@0.21.5/node_modules/esbuild-register/dist/node.js:4838:24)
```

storybook-react-rsbuild does not limit storybook versions, so cannot downgrade storybook-react-rsbuild's dependency version now.

**Comment by yf-yang** [comment]
https://github.com/storybookjs/storybook/blob/a997fa100dac2efe648515c63c23c357378c120c/code/builders/builder-webpack5/src/preview/iframe-webpack.config.ts#L12

**Comment by yf-yang** [comment]
Those are breaking changes :/

@storybook/types -> storybook/internal/types
@storybook/preview/globals -> storybook/internal/preview/globals
@storybook/core-common -> storybook/internal/common

I'm wondering how you'd like to fix it, since Storybook's builder's version is aligned with storybook version, but storybook-rsbuild's version is not aligned. Will you support different versions simultaneously?

**Comment by yf-yang** [comment]
Cannot easily fix it by patching and replacing import paths. @fi3ework Would you please limit storybook-rsbuild's dependency to ~8.1.11?

**Comment by Atmos4** [comment]
Local fix for me: `pnpm add -D @storybook/preview@8.1.11`

Seems like they messed that up in 8.2.

**Comment by yf-yang** [comment]
Oh, thank you @Atmos4! Nice catch 👍. At least it could work now.

**Comment by yf-yang** [comment]
@Atmos4 Seems preview's global settings (parameters, loaders, etc) are still broken? Do you observe similar stuff?

**Comment by fi3ework** [comment]
> Those are breaking changes :/
> 
> @storybook/types -> storybook/internal/types @storybook/preview/globals -> storybook/internal/preview/globals @storybook/core-common -> storybook/internal/common

Thanks for sharing the clue, I'm fixing the issue.

> I'm wondering how you'd like to fix it, since Storybook's builder's version is aligned with storybook version, but storybook-rsbuild's version is not aligned. Will you support different versions simultaneously?

NO, we only may maintain a legacy and a new coming version when Storybook bumping its major version. For minor version changes, we will limit the minimal version of storybook alongisde.

**Comment by fi3ework** [comment]
Not sure will #37 one hundred makes up the updates. Feel free to reopen the issue if there's still 8.2.0 related bug after the version released.

**Comment by yf-yang** [comment]
Sure. Thank you for your prompt response! Take care of your health! May I ask when the next version will be released?

**Comment by fi3ework** [comment]
Later in today.

**Comment by fi3ework** [comment]
Fixed in 0.0.7.
